### ISSUE TITLE
Transaction error in datastore returning odd and not friendly response

### ISSUE THREAD

**Issue by stephenplusplus** [googleapis/nodejs-datastore#8]
###### Copied from original issue: https://github.com/GoogleCloudPlatform/google-cloud-node/issues/2106

<a href="/susanlinsfu"><img src="https://avatars2.githubusercontent.com/u/15883441?s=88&v=4" height=44 width=44 align=left></a><a href="/susanlinsfu">@&shy;susanlinsfu</a><br>March 19, 2017 8:27 PM

#### Environment details

  - OS: Ubuntu 16.04
  - Node.js version: 7.7.3
  - npm version: 4.1.2
  - google-cloud-node version: 0.49

I insert the following entity, but it will fail because the entity already exists:
```js
var transaction = datastore.transaction();
                    transaction.run(function(err) {
                        if (err) {
                            // Error handling omitted.
                            return callback('Database Error');
                        }

                        transaction.save([{
                            key: datastore.key(mykind, mykey]),
                            method: 'insert',
                            data: [{
                                name: 'par1',
                                value: val1
                            }, {
                                name: 'par2',
                                value: val2,
                                excludeFromIndexes: true
                            }
                        }]);

                        transaction.commit(function(err) {
                            if (!err) {
                                // Transaction committed successfully.
                                return callback(null, 'Everything is cool!');
                            }
                             console.log(err);
                        });
                    });
```

The error output is very hard to parse. Err returns this:
`
"{\"error\":\"Error: entity already exists: app: \\\"s~someapp\\\"<br/>path <<br/>  Element {<br/>    type: \\\"mykind\\\"<br/>    name: \\\"mykey\\\"<br/>  }<br/>><br/>\"}"`

It returns javascript object but it is difficult to parse if you need the error, type, and name fields. It is not in proper format and there are html entities in the response.

**Comment by stephenplusplus** [comment]
<a href="/lukesneeringer"><img src="https://avatars2.githubusercontent.com/u/4346?s=88&v=4" height=44 width=44 align=left></a>__<a href="/lukesneeringer">@&shy;lukesneeringer</a>__<br>March 20, 2017 4:16 PM

Hello @susanlinsfu, thanks for reporting. I agree this is not a good error, and will see what I can do about it.

**Comment by stephenplusplus** [comment]
<a href="/susanlinsfu"><img src="https://avatars2.githubusercontent.com/u/15883441?s=88&v=4" height=44 width=44 align=left></a>__<a href="/susanlinsfu">@&shy;susanlinsfu</a>__<br>April 20, 2017 10:56 PM

@lukesneeringer Hi Luke, thanks for looking into the issue. I have an application with a deadline coming up this week which I need this feature to work (being able to parse the error message). I understand there are other issues and before this fix gets officially released it could be months. I have looked into the source code for transactions.js, but could not find where the error was being generated. I was wondering if you could informally just point me to where the error is being generated so I can modify this in my application within my own project? Cheers

**Comment by stephenplusplus** [comment]
###### _November 2, 2017 3:43 PM_

@lukesneeringer this would either require a change in the Datastore API or our stripping the HTML tags from the response using something like [strip](http://gitnpm.com/strip):

```js
var errorMessage = 'entity already exists: app: "s~nth-circlet-705"<br/>path <<br/>  Element {<br/>    type: "Company"<br/>    name: "Google"<br/>  }<br/>><br/>'
require('strip')(errorMessage).replace(/\s+/g, ' ')
// 'entity already exists: app: "s~nth-circlet-705"path < Element { type: "Company" name: "Google" }>'
```

What do you think?

**Comment by callmehiphop** [comment]
This is definitely still an issue, although I don't think its one for the client to solve. I'm going to close this and suggest a new issue get opened over at https://cloud.google.com/support/docs/issue-trackers if this is still something people want.
